pandas/tools/tests/test_merge_asof.py

```diff
@@ -221,6 +221,117 @@ def test_missing_right_by(self):
         expected.loc[expected.ticker == 'MSFT', ['bid', 'ask']] = np.nan
         assert_frame_equal(result, expected)
 
+    def test_multiby(self):
```

Comment:
can you add a test for the raising on an invalid merge type (I think only float16?)